pandas/tests/indexes/test_interval.py

```diff
@@ -348,9 +355,126 @@ def test_get_loc_value(self):
+        pytest.raises(KeyError, left.get_loc, Interval(1, 2, closed='both'))
+        pytest.raises(KeyError, left.get_loc, Interval(2, 3, closed='both'))
+        pytest.raises(KeyError, left.get_loc, Interval(3, 4, closed='both'))
+        pytest.raises(KeyError, left.get_loc, Interval(0, 2, closed='both'))
+        pytest.raises(KeyError, left.get_loc, Interval(2.5, 3, closed='both'))
+        pytest.raises(KeyError, left.get_loc, Interval(-1, 4, closed='both'))
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_get_loc_value_closed_both_updated_behavior(self):
+
+        both = IntervalIndex.from_tuples([(0, 1), (2, 3)], closed='both')
+
+        pytest.raises(KeyError, both.get_loc, -0.5)
+        assert both.get_loc(0)   == 0
+        assert both.get_loc(0.5) == 0
+        assert both.get_loc(1)   == 0
+        pytest.raises(KeyError, both.get_loc, 1.5)
+        assert both.get_loc(2)   == 1
+        assert both.get_loc(2.5) == 1
+        assert both.get_loc(3)   == 1
+        pytest.raises(KeyError, both.get_loc, 3.5)
+
+        pytest.raises(KeyError, both.get_loc, Interval(0, 1, closed='right'))
+        pytest.raises(KeyError, both.get_loc, Interval(1, 2, closed='right'))
+        pytest.raises(KeyError, both.get_loc, Interval(2, 3, closed='right'))
+        pytest.raises(KeyError, both.get_loc, Interval(3, 4, closed='right'))
+        pytest.raises(KeyError, both.get_loc, Interval(0, 2, closed='right'))
+        pytest.raises(KeyError, both.get_loc, Interval(2.5, 3, closed='right'))
+        pytest.raises(KeyError, both.get_loc, Interval(-1, 4, closed='right'))
+
+        pytest.raises(KeyError, both.get_loc, Interval(0, 1, closed='left'))
+        pytest.raises(KeyError, both.get_loc, Interval(1, 2, closed='left'))
+        pytest.raises(KeyError, both.get_loc, Interval(2, 3, closed='left'))
+        pytest.raises(KeyError, both.get_loc, Interval(3, 4, closed='left'))
+        pytest.raises(KeyError, both.get_loc, Interval(0, 2, closed='left'))
+        pytest.raises(KeyError, both.get_loc, Interval(2.5, 3, closed='left'))
+        pytest.raises(KeyError, both.get_loc, Interval(-1, 4, closed='left'))
+
+        assert both.get_loc(Interval(0, 1, closed='both')) == 0
+        pytest.raises(KeyError, both.get_loc, Interval(1, 2, closed='both'))
+        assert both.get_loc(Interval(2, 3, closed='both')) == 1
+        pytest.raises(KeyError, both.get_loc, Interval(3, 4, closed='both'))
+        pytest.raises(KeyError, both.get_loc, Interval(0, 2, closed='both'))
+        pytest.raises(KeyError, both.get_loc, Interval(2.5, 3, closed='both'))
+        pytest.raises(KeyError, both.get_loc, Interval(-1, 4, closed='both'))
+
     def slice_locs_cases(self, breaks):
         # TODO: same tests for more index types
-        index = IntervalIndex.from_breaks([0, 1, 2], closed='right')
+        index = IntervalIndex.from_breaks(breaks, closed='right')
```

Comment:
This is a good fix (the tests here were actually not testing anything), but now this is failing the `test_slice_locs_decreasing_int64/float64` test, because they don't pass the breaks as a list, but as tuples (so would need a `from_tuples`. So maybe easiest is to change all to tuples? (because with breaks you cannot have overlapping intervals)